### PR TITLE
Use mem::take instead of mem::replace where appropriate

### PR DESCRIPTION
**Pull request by NikVolf**

### CODE REVIEW

**Comment by arkpar** [comment]
`mem::take` was stabilized in 1.40.0. Not sure if this is old enough to be set as our minimum supported compiler version.

**Comment by bkchr** [comment]
We already require at least `1.42.0`

File: client/api/src/cht.rs [hunk 1]
```diff
@@ -232,7 +232,7 @@ pub fn for_each_cht_group<Header, I, F, P>(
 			functor_param = functor(
 				functor_param,
 				current_cht_num,
-				::std::mem::replace(&mut current_cht_blocks, Vec::new()),
+				::std::mem::take(&mut current_cht_blocks),
```
Comment (bkchr) [inline client/api/src/cht.rs:235]:
```suggestion
				std::mem::take(&mut current_cht_blocks),
```

File: client/api/src/cht.rs [hunk 2]
```diff
@@ -244,7 +244,7 @@ pub fn for_each_cht_group<Header, I, F, P>(
 		functor(
 			functor_param,
 			current_cht_num,
-			::std::mem::replace(&mut current_cht_blocks, Vec::new()),
+			::std::mem::take(&mut current_cht_blocks),
```
Comment (bkchr) [inline client/api/src/cht.rs:247]:
```suggestion
			std::mem::take(&mut current_cht_blocks),
```

**Comment by arkpar** [comment]
bot merge

**Pull request merged**